### PR TITLE
setup: loosen version pinning

### PR DESCRIPTION
**Pull request by oliver-sanders**
I think we are being overly specific with dependency versions, particularly for testing.

We now have conflicting dependencies between different Cylc repositories which causes pip to install dependencies at undesirable versions which is worse than not version pinning at all.

Today's annoying conflict was between the version of pycodestyle used be Cylc-flow and flake8.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,17 +51,16 @@ def find_version(*file_paths):
     'markupsafe==1.1.*',
```
Comment (oliver-sanders) [inline setup.py:51]: I don't think we are using this module, is it listed here to allow us to pin a dependency of a dependency?

File: setup.py [hunk 2]
```diff
@@ -51,17 +51,16 @@ def find_version(*file_paths):
...
     'protobuf==3.12.0rc1',
     'pyzmq==18.1.*',
-    'click>=7.0',
```
Comment (oliver-sanders) [inline setup.py:54]: Duplicate.

Comment (kinow) [reply setup.py:54]: Nice catch

File: setup.py (hunk 1 above)
Comment (kinow) [reply setup.py:51]: I **think** we used to have this in the old `lib` folder of Cylc... and I just copied it to setup.py (trying to remember)

Comment (oliver-sanders) [reply setup.py:51]: I think you are right there.

Comment (oliver-sanders) [reply setup.py:51]: It may have been to do with the old GUI.

Comment (kinow) [reply setup.py]: Feel free to remove if not used :+1: this change looks good, approving assuming CI passes.

Comment (oliver-sanders) [reply setup.py]: I pip uninstalled and the unittests still passed so I think we are good to drop it.

**Pull request merged**